### PR TITLE
Track game retention shares

### PR DESCRIPTION
**Pull request by austinhallock**

### CODE REVIEW

File: src/root.coffee [hunk 1]
```diff
@@ -66,6 +67,9 @@ ga? 'set', 'dimension1', if EnvironmentService.isClayApp() then 'clay_app' \
                          else if EnvironmentService.isKikEnabled() then 'kik' \
                          else if EnvironmentService.isAndroid() then 'android' \
                          else 'web'
+KikService.isFromPush().then (isFromPush) ->
+  if isFromPush
+    ga? 'set', 'referrer', 'http://push.kik.com'
```
Comment (Zolmeister) [inline src/root.coffee]: find a url that wont consflict

File: src/root.coffee [hunk 2]
```diff
@@ -66,6 +67,9 @@ ga? 'set', 'dimension1', if EnvironmentService.isClayApp() then 'clay_app' \
                          else if EnvironmentService.isKikEnabled() then 'kik' \
                          else if EnvironmentService.isAndroid() then 'android' \
                          else 'web'
+KikService.isFromPush().then (isFromPush) ->
```
Comment (Zolmeister) [inline src/root.coffee]: race

**Comment by austinhallock** [comment]
updated

File: src/root.coffee [hunk 3]
```diff
@@ -66,6 +67,10 @@ ga? 'set', 'dimension1', if EnvironmentService.isClayApp() then 'clay_app' \
                          else if EnvironmentService.isKikEnabled() then 'kik' \
                          else if EnvironmentService.isAndroid() then 'android' \
                          else 'web'
+setReferrerPromise = KikService.isFromPush().then (isFromPush) ->
```
Comment (Zolmeister) [inline src/root.coffee]: garbage
move this down to where the other kik and analytics code is

**Comment by austinhallock** [comment]
how does this look? https://github.com/claydotio/clay-mobile/commit/1011a1091d00e4a93b778b539396a22393474261

**Comment by austinhallock** [comment]
https://github.com/claydotio/clay-mobile/commit/2d0c09ad5706996584e99e0a497dbcf2894d51c8

**Comment by austinhallock** [comment]
tests, sec

**Comment by austinhallock** [comment]
updated / rebased

File: src/services/kik.coffee [hunk 4]
```diff
@@ -10,50 +9,15 @@ class KikService
   constructor: ->
     @isFromPushPromise = null
 
-  isFromPush: =>
+  isFromPush: ->
```
Comment (Zolmeister) [inline src/services/kik.coffee]: now it's just broken.

Comment (austinhallock) [inline src/services/kik.coffee]: Updated

**Comment by Zolmeister** [comment]
rebase

**Comment by austinhallock** [comment]
done

**Pull request merged**